src/store/ingredient.ts

```diff
@@ -42,13 +42,13 @@ export const $updateSelectedIngredients = atom(
 export const $selectedIngredient = atom<IngredientKey | null>(null);
 export const $updateSelectedIngredient = atom(
   (get) => get($selectedIngredient),
-  (get, set, selectedIngredient: IngredientKey | []) => {
-    if (Array.isArray(selectedIngredient)) {
-      set($selectedIngredients, selectedIngredient);
+  (get, set, selectedIngredient: IngredientKey | null) => {
+    if (selectedIngredient === null) {
+      set($selectedIngredient, null);
       return;
     }
 
-    if (selectedIngredient === get($selectedIngredient)) {
+    if (get($selectedIngredient) === selectedIngredient) {
```

Comment:
```suggestion
    if (selectedIngredient === null || get($selectedIngredient) === selectedIngredient) {
```

바로 위 If문이랑 조거문 합쳐줄 수 있지 않을가요?